exampleCourse/courseInstances/Sp15/assessments/this-should-fail

```diff
@@ -0,0 +1 @@
+testing!
```

Comment:
Oops, this was left over from when I was iterating on the new code to catch files when loading `info.json` from a directory. Will remove (this should not fail a sync, btw).

Reply:
I think this used to throw an error with the old sync code? That is, we ignore files, but if there is a directory then it’s an error for it to not contain an `info*.json` file. The original intention with this was to avoid the case where someone accidentally renamed something like an `infoCourseInstance.json` file (perhaps just a single letter change), which would cause us to delete the entire course instance (this is recoverable, but it might be a bit scary 😄).

I’m not convinced that we need to preserve this behavior. If we did want to do it then we could always `lstat()` only on failed `open()`, which should still be fast.

What do you think? Is this just worrying about an edge case that’s very unlikely in practice?

Reply:
Ah, that's a different case than what I was testing with this file. If you look in `lib/course-db.js`, I'm careful to only handle `ENOTDIR` which is what we'll get if we try to read a file from a directory that doesn't exist (which is what would happen if there was a text file in`questions`, for instance). `ENOENT` for a missing `info.json` file still counts as an error. I'll write test cases for these too 🙂